test/unit/fixture/reverse_with_layer_filtering_non_coarse_subset.js

```diff
@@ -0,0 +1,41 @@
+var vs = require('../../../query/reverse_defaults');
+
+module.exports = {
+  'query': {
+    'bool': {
+      'filter': [
+        {
+          'geo_distance': {
+            'distance': '500km',
```

Comment:
Our previous experiences with slow reverse queries suggest that anything that can query over such a large distance has the potential to be very slow. Just to be sure, this 500km is only here because of the `boundary.radius` set in the test, right? Even so, we might want to lower it. Even setting the example of a 500km radius search for all venues and streets could cause us to write non-performant code someday.